streaming/java/streaming-api/src/main/java/org/ray/streaming/api/context/StreamingContext.java

```diff
@@ -48,8 +48,12 @@ public void execute() {
     this.plan = planBuilder.buildPlan();
     plan.printPlan();
 
-    IJobSchedule jobSchedule = new JobScheduleImpl(jobConfig);
-    jobSchedule.schedule(plan);
+    ServiceLoader<IJobSchedule> serviceLoader = ServiceLoader.load(IJobSchedule.class);
+    Iterator<IJobSchedule> iterator = serviceLoader.iterator();
+    Preconditions.checkArgument(iterator.hasNext());
+    // IJobSchedule jobSchedule = new JobScheduleImpl(jobConfig);
+    IJobSchedule jobSchedule = iterator.next();
```

Comment:
how about save the call of iterator.hasNext() and replace this line of code with the following code block:

 try {
            iterator.next();
        } catch (NoSuchElementException e) {
            throw new IllegalArgumentException("No JobScheduler implementation instance has been configured.");
        }

Reply:
fixed

Reply:
Added an error message in `Preconditions.checkArgument`